packages/application-components/src/components/portals-container/portals-container.tsx

```diff
@@ -129,11 +129,15 @@ const PortalsContainer = forwardRef<TLayoutRefs, TPortalsContainerProps>(
     // The stacking layers are then re-calculated.
     useMutationObserver(
       portalsContainerRef,
-      (mutation) => {
+      (mutationList) => {
+        // We expect only a single element in the mutation list as we configured the
+        // observer to only listen to `data-theme` changes.
```

Comment:
**nit** It seems we're not listening to `data-theme` changes here but `childList` changes.